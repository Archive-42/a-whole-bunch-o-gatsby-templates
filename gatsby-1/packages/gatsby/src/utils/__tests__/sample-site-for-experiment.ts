import sampleSiteForExperiment from "../sample-site-for-experiment"

import {
  getRepositoryId,
  IRepositoryId,
} from "gatsby-telemetry/lib/repository-id"
;((getRepositoryId as unknown) as jest.Mock<IRepositoryId>).mockReturnValue({
  repositoryId: `mockId`,
})

jest.mock(`gatsby-telemetry/lib/repository-id`)

describe(`sampleSiteForExperiment`, () => {
  it(`returns true or false depending on if they randomly are bucketed in or not`, () => {
    const experiments = [`a`, `b`, `c`, `d`, `e`, `f`, `h`, `i`]
    experiments.forEach(experiment => {
      expect(sampleSiteForExperiment(experiment, 0)).toBeFalsy()
      expect(sampleSiteForExperiment(experiment, 50)).toMatchSnapshot(
        experiment
      )
      expect(sampleSiteForExperiment(experiment, 100)).toBeTruthy()
    })
  })
})
