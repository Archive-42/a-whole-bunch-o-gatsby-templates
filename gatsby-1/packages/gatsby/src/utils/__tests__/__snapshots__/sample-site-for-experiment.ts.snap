// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sampleSiteForExperiment returns true or false depending on if they randomly are bucketed in or not: a 1`] = `true`;

exports[`sampleSiteForExperiment returns true or false depending on if they randomly are bucketed in or not: b 1`] = `false`;

exports[`sampleSiteForExperiment returns true or false depending on if they randomly are bucketed in or not: c 1`] = `false`;

exports[`sampleSiteForExperiment returns true or false depending on if they randomly are bucketed in or not: d 1`] = `true`;

exports[`sampleSiteForExperiment returns true or false depending on if they randomly are bucketed in or not: e 1`] = `true`;

exports[`sampleSiteForExperiment returns true or false depending on if they randomly are bucketed in or not: f 1`] = `false`;

exports[`sampleSiteForExperiment returns true or false depending on if they randomly are bucketed in or not: h 1`] = `false`;

exports[`sampleSiteForExperiment returns true or false depending on if they randomly are bucketed in or not: i 1`] = `false`;
